### PR TITLE
Fix#8577: Refactor part of the secrets manager implementation 

### PR DESCRIPTION
**Pull request by nahuelverdugo**
Part of: #8577

### Describe your changes :
- Removed part of the old implementation of the secrets manager
- Added a new `pydantic` class for handling the secrets string
- Change how the connections params and options are handled

### Type of change :
- [x] Improvement

### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### Reviewers
Backend: @open-metadata/backend
Ingestion: @open-metadata/ingestion

### CODE REVIEW

File: ingestion/src/metadata/clients/aws_client.py [hunk 1]
```diff
@@ -24,22 +25,40 @@
     QuickSightClient,
     SageMakerClient,
 )
-from metadata.generated.schema.security.credentials.awsCredentials import AWSCredentials
 from metadata.utils.logger import utils_logger
 
 logger = utils_logger()
 
 
+class AWSCredentials(BaseModel):
```
[the remark this answers is not in the document]
Comment (nahuelverdugo) [reply ingestion/src/metadata/clients/aws_client.py:33]: This is a tricky one, if I use the generated models I create a circular dependency where: 
`AWSCredentials` -> `SecretStr` -> `SecretManagerFactory` -> `AWSCredentials`
I didn't come up with anything else to bypass this issue but I am open to ideas on how to handle it.

File: ingestion/src/metadata/utils/source_connections.py [hunk 2]
```diff
@@ -88,10 +88,44 @@
     VerticaConnection,
 )
 from metadata.generated.schema.security.credentials.gcsCredentials import GCSValues
+from metadata.ingestion.models.custom_pydantic import CustomSecretStr
 
 CX_ORACLE_LIB_VERSION = "8.3.0"
 
 
+def update_connection_opts_args(connection):
```
[the remark this answers is not in the document]
Comment (nahuelverdugo) [reply ingestion/src/metadata/utils/source_connections.py:96]: Nice to read I was not the only one who thought the same! :D

File: ingestion/src/metadata/ingestion/api/workflow.py [hunk 3]
```diff
@@ -107,8 +102,6 @@ def __init__(
 
         self._retrieve_service_connection_if_needed(service_type)
 
-        self._retrieve_dbt_config_source_if_needed(service_type)
```
[the remark this answers is not in the document]
Comment (nahuelverdugo) [reply ingestion/src/metadata/ingestion/api/workflow.py:110]: We don't need that anymore with these changes. Now if there is something in the DBT config marked as a password and the password format is `secret:secret-id` the `CustomSecretStr` will search for it in the secret store when calling `get_secret_value()`.

File: ingestion/src/metadata/clients/aws_client.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nahuelverdugo) [reply ingestion/src/metadata/clients/aws_client.py]: Fixed after discussing it offline. With a local import, the issue is avoided.

**Review by pmbrull — APPROVED** [review]

**Pull request merged**